tests/jobs/test_cli.py

```diff
@@ -74,15 +74,33 @@ def test_create_cli_json_file(jobs_api_mock, tmpdir):
 
 RESET_JSON = '{"job_name": "test_job"}'
 
+CORRECT_REQUEST_PAYLOAD = {'job_id': 1,
+                           'new_settings': json.loads(RESET_JSON)}
+
 
 @provide_conf
 def test_reset_cli_json(jobs_api_mock):
     runner = CliRunner()
     runner.invoke(cli.reset_cli, ['--json', RESET_JSON, '--job-id', 1])
-    assert jobs_api_mock.reset_job.call_args[0][0] == {
-        'job_id': 1,
-        'new_settings': json.loads(RESET_JSON)
-    }
+    assert jobs_api_mock.reset_job.call_args[0][0] == CORRECT_REQUEST_PAYLOAD
+
+
+RESET_JSON_NEW_SETTINGS = '{ "new_settings": %s, "job_id": 1 }' % RESET_JSON
+
+
+@provide_conf
+def test_reset_cli_json_new_settings(jobs_api_mock):
+    runner = CliRunner()
+    runner.invoke(cli.reset_cli, [
+                  '--json', RESET_JSON_NEW_SETTINGS, '--job-id', 1])
```

Comment:
This doesn't test if the argument takes precedence or not. It has the same behavior as the test below so both would pass even if you ignore the command line argument altogether. Could you update this to use a different job ID to ensure it doesn't use the job ID from the JSON payload?